pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -1226,6 +1226,40 @@ func (c *Controller) WorkloadInstanceHandler(si *model.WorkloadInstance, event m
 		c.workloadInstancesIndex.Insert(si)
 	}
 
+	// handle WorkloadInstance in ambient by creating a mock pod that models the VM workload.
+	ambientMockPod := &v1.Pod{
```

Comment:
Suggest making this handling in ambientindex.go, maybe add a new function AmbientIndex.handleWorkloadInstance